crates/wit-component/src/validation.rs

```diff
@@ -425,7 +425,7 @@ fn validate_exported_interface(
                     wasm_sig_to_func_type(interface.wasm_signature(AbiVariant::GuestExport, f));
                 let ty = types.function_at(*func_index).unwrap();
                 if ty == &expected_ty {
-                    return Ok(());
+                    continue;
```

Comment:
Oops thanks for catching this!

Reply:
Thanks goes to `clippy` as my rust-analyzer check command :)